doc/netplan.md

```diff
@@ -340,6 +340,90 @@ These options are available for all types of interfaces.
      :    Match this policy rule based on the type of service number applied to
           the traffic.
 
+## Authentication
+Netplan supports advanced authentication settings for ethernet and wifi
+interfaces, as well as individual wifi networks, by means of the ``auth`` block.
+
+``auth`` (mapping)
+
+:    Specifies authentication settings for a device of type ``ethernets:`` or
+     ``wifis:``, or a single wifi ``access-points:`` entry. If ``auth`` blocks
+     are specified on both a wifi device and one of its access points, only the
+     settings in the access point's ``auth`` block are taken into account;
+     however, the ``auth`` block of the wifi device, if present, is used for any
+     ``access-points:`` without their own ``auth`` block.
+
+     Example:
+
+          wifis:
+            wlp1s0:
+              auth:
+                key-mgmt: wpa-psk
+                psk: "d3f4ul7"
+              access-points:
+                office:
+                  auth:
+                    # the authentication settings above are ignored for the
+                    # wifi named "office"
+                    key-mgmt: wpa-eap
+                    eap-method: ttls
+                    anonymous-identity: "@internal.example.com"
+                    identity: "joe@internal.example.com"
+                    password: "v4ryS3kr1t"
+                home:
+                  mode: infrastructure
+                  # the authentication settings above are applied to the wifi
+                  # named "home"
+
+      The ``auth`` block supports the following properties:
+
+      ``key-mgmt`` (scalar)
+      :    The supported key management modes are ``none`` (no key management),
+           ``wpa-psk`` (WPA with pre-shared key, common for home wifi),
+           ``wpa-eap`` (WPA with EAP, common for enterprise wifi) and ``8021x``
```

Comment:
Let's use "802.1x" as the value. For a scalar value, it's perfectly fine to use and should be easier to figure out than whether one should write "8021x" or "802-1x" -- at least, it's exactly the name of the standard.

Similarly, what about dropping "wpa-" from the other names? Would that work and still avoid clashing with other key-management methods?

Reply:
Dropping "wpa-" might become a problem if NM starts supporting additional key management schemes that wpa_supplicant [already supports](https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_supplicant.conf?id=0fe3ede0a4f6fe984bbaad90f1a9d9e8decc155d#n870). Not sure how likely that is, though.

Reply:
That's true of FT-*, but I think we'll be fine calling those whatever value they are exactly, whereas any option starting with "WPA-" we'd be better off not having "WPA-" in the name -- it's shorter, and we don't need that bit -- it's already obvious that it's wpa. In my mind, people are more likely to know, without looking at doc, to write "psk" or "eap", and wouldn't know that they'd need to name those "wpa-psk", "wpa-eap" without looking at the doc.

I'm not strongly attached to either way, but I think anything we can do to make "looking at the doc" less necessary to configure some of the "simplest" complex network configurations (like WPA) is a win.